trading_calendars/trading_calendar.py

```diff
@@ -1051,6 +1174,47 @@ def _overwrite_special_dates(midnight_utcs,
     opens_or_closes.values[indexer] = special_opens_or_closes.values
 
 
+def _remove_breaks_for_special_dates(
+    midnight_utcs, break_start_or_end, special_opens_or_closes
+):
+    """
+    Overwrite breaks in break_start_or_end with corresponding dates in
+    special_opens_or_closes, using midnight_utcs for alignment.
+    """
+    # Short circuit when we have no breaks
+    if break_start_or_end is None:
+        return
+
+    # Short circuit when nothing to apply.
+    if not len(special_opens_or_closes):
+        return
+
+    len_m, len_oc = len(midnight_utcs), len(break_start_or_end)
+    if len_m != len_oc:
+        raise ValueError(
+            "Found misaligned dates while building calendar.\n"
+            "Expected midnight_utcs to be the same length as break_starts,\n"
+            "but len(midnight_utcs)=%d, len(break_start_or_end)=%d"
+            % len_m, len_oc
+        )
+
+    # Find the array indices corresponding to each special date.
+    indexer = midnight_utcs.get_indexer(special_opens_or_closes.index)
+
+    # -1 indicates that no corresponding entry was found.  If any -1s are
+    # present, then we have special dates that doesn't correspond to any
+    # trading day.
+    if -1 in indexer:
+        bad_dates = list(special_opens_or_closes[indexer == -1])
+        raise ValueError("Special dates %s are not trading days." % bad_dates)
+
+    # NOTE: This is a slightly dirty hack.  We're in-place overwriting the
+    # internal data of an Index, which is conceptually immutable.  Since we're
```

Comment:
Is it important that we mutate this index, instead of creating a new one?

Reply:
This copies the logic we use in `_overwrite_special_dates`